vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -152,16 +153,25 @@ func filterNonEvictablePods(pods []*apiv1.Pod, evictionRestriciton eviction.Pods
 	return result
 }
 
-func filterDeletedPods(pods []*apiv1.Pod) []*apiv1.Pod {
+func filterNotLivePods(pods []*apiv1.Pod) []*apiv1.Pod {
 	result := make([]*apiv1.Pod, 0)
 	for _, pod := range pods {
-		if pod.DeletionTimestamp == nil {
+		if isPodLive(pod) {
 			result = append(result, pod)
 		}
 	}
 	return result
 }
 
+func isPodLive(pod *apiv1.Pod) bool {
+	if pod == nil {
```

Comment:
I can remove, but I was planning to extract this method do common utils (I remember I have one more place I need to use this) and then it would make sense to keep the check.